vlfs/src/driver/flash.rs

```diff
@@ -1,73 +1,133 @@
     async fn write_256b<'b>(
+        /*
+        Purpose: write 256 bytes of data to the memory device starting at the specified address
+        Parameters:
+            &mut self:     a reference to the memory device object
+            address:       u32 integer that specifies the starting address of the 256 byte block that will be written
+            write_buffer:  buffer where data to be written is stored
+        
+        Outputs:
+            Ok(()):        returns nothing if the write operation was successful
+            Err(Self::Error): returns error that occured
+
+        Note:
+            size of the buffer must be at least 261 bytes long */
+
         &mut self,
         address: u32,
         write_buffer: &'b mut [u8],
     ) -> Result<(), Self::Error>;
 
-    // read arbitary length, length of read_buffer must be larger or equal to read_length + 5
+    //read arbitrary length, length of read_buffer must be larger or equal to read_length + 5
+    //refer to the read_4kib function for more details on the parameters and outputs as they are the same
     async fn read<'b>(
         &mut self,
         address: u32,
         read_length: usize,
         read_buffer: &'b mut [u8],
     ) -> Result<&'b [u8], Self::Error> {
+        /*
+        This loop reads the data from the flash by calling read_4kib multiple times, with each call
+        reading up to 4096 bytes at a time, until it has read the entire requested length.
+
+        Variables:
+            bytes_read: usize integer that keeps track of the number of bytes that have been read so far
+            length:     usize integer that specifies the number of bytes to be read in the current iteration
+         */
+
         let mut bytes_read = 0;
         while bytes_read < read_length {
+            // Determine the maximum number of bytes to read in this iteration, which is either the remaining
+            // number of bytes to read or the maximum of 4096.
             let length = if read_length - bytes_read > 4096 {
                 4096
             } else {
                 read_length - bytes_read
             };
-            // info!("reading {}/{} bytes", bytes_read, read_length);
+            // Call read_4kib with the current address and length to read, and store the result in the read_buffer
+            // at the appropriate location.
             self.read_4kib(
                 address + bytes_read as u32,
                 length,
                 &mut read_buffer[bytes_read..],
             )
             .await?;
+            // Update the number of bytes read.
             bytes_read += length;
         }
 
+        // Return the slice of the read_buffer that contains the actual data. This is done by taking a slice of the read_buffer
+        // from the 5th byte to the 5th byte + the read_length. The first 5 bytes are skipped because they are used to store the
+        // address and length of the data that was read.
```

Comment:
> The first 5 bytes are skipped because they are used to store the address and length of the data that was read.

This is inaccurate, the first 5 bytes are used for the spi command, see https://github.com/McMaster-Rocketry-Team/VLF3-firmware/blob/ce9025236d2b7b10305f6a291b5b1bf532516db1/src/flash.rs#LL205C31-L205C31